### PR TITLE
[diffusion] Add hierarchical startup-time breakdown (SGLANG_DIFFUSION_STARTUP_PROFILE)

### PR DESCRIPTION
**Pull request by zetyquickly**
## Summary

- add an opt-in hierarchical startup timer behind `SGLANG_DIFFUSION_STARTUP_PROFILE`, a plain pass-through when disabled
- instrument the launch path: device set, distributed init, `build_pipeline`, pipeline instantiation, and per-component weight loading
- report each phase as a percentage of its immediate parent, and log the tree once from rank 0
- add unit coverage for nesting, parent-relative percentages, and the disabled no-op path

## Motivation

#19087 asked for a detailed breakdown of diffusion launch time. The investigation there was done with ad-hoc print timing that was never committed, so the next person to look at launch cost starts from scratch. This lands the breakdown as a permanent facility.

The instrumented boundaries follow that thread's own findings: component loading dominates (Qwen-Image ~63s of a ~77s launch, split text_encoder / transformer), so `load_component.{name}` is the highest-signal level and is broken out per component.

## Modifications

- new `runtime/utils/startup_profiler.py`: `StartupProfiler.phase()` nests timings into a tree; `render()` flattens to dotted paths (`load_modules.load_component.text_encoder: 32960.00ms (52.0%)`). When disabled, `phase()` yields immediately — no node, no timer.
- new env var `SGLANG_DIFFUSION_STARTUP_PROFILE` in `envs.py`, following the existing `SGLANG_DIFFUSION_STAGE_LOGGING` pattern.
- `gpu_worker.py`: wrap `set_device`, `init_distributed_environment`, `build_pipeline`; emit the summary after init.
- `pipelines_core/__init__.py`: wrap `get_model_info`, `pipeline_instantiation`.
- `composed_pipeline_base.py`: wrap `load_modules`, `initialize_pipeline`, `create_pipeline_stages`, and each `load_component.{module_name}`.

Percentages are parent-relative rather than total-relative so a nested phase reads as "share of its parent" — matching the format used in the issue thread.

## Accuracy Tests

Not applicable — instrumentation only. No model code, numerics, or execution order changes; the profiler wraps existing calls in a context manager and is inert unless the env var is set.

## Speed Tests and Profiling

Disabled (default): `phase()` is a bare `yield` — no tree allocation, no `perf_counter` call. Enabled: a handful of `perf_counter` calls around phases that already take seconds.

## Validation

- `python -m py_compile` clean on all changed files
- `black --check` and `isort --check-only` (repo config, `.isort.cfg`) clean on all 6 changed files
- profiler logic verified standalone (`3 passed`): disabled-is-noop, percentages relative to immediate parent, sibling top-level phases do not nest
- rank gating: summary emitted only when `get_is_main_process()`, so an 8-GPU launch logs one tree rather than eight

Not yet run: an end-to-end launch of a real diffusion server with the flag enabled. I do not have NVIDIA hardware for this, and the AMD box used for the sibling PR became unreachable. The example output in the issue thread is what the format is modelled on, not a capture from this branch — happy to attach a real capture if a reviewer can run it, or to hold the PR until I can.

## Checklist

- [x] Add unit tests according to the contribution guide — `test/unit/test_startup_profiler.py`
- [x] Format code with pre-commit — `black --check` / `isort --check-only` clean; did not run the full pre-commit hook set (codespell, clang-format, etc.)
- [x] Follow the SGLang code style guidance

Refs #19087 (breakdown/instrumentation only; the follow-up optimizations discussed in the thread — parallel component loading, pinned-memory transfer, skipping unnecessary `nn.Embedding` init — are separate changes, and #19090 tracks sleep/wake)

---
### CI States

Latest PR Test (Base): :x: [Run #31975208837](https://github.com/sgl-project/sglang/actions/runs/31975208837)
Latest PR Test (Extra): :x: [Run #31975208619](https://github.com/sgl-project/sglang/actions/runs/31975208619)

### CODE REVIEW

**Comment by zetyquickly** [comment]
Real capture from an actual launch (Wan2.1-T2V-1.3B, AMD Instinct MI300X VF, 1 GPU, ROCm 7.2 / gfx942, `lmsysorg/sglang:v0.5.14-rocm720-mi30x` base image with this branch's `python/sglang` overlaid):

```
SGLANG_DIFFUSION_STARTUP_PROFILE=1 sglang generate \
  --model-path Wan-AI/Wan2.1-T2V-1.3B-Diffusers \
  --prompt "A curious raccoon" \
  --save-output
```

```
[08-16 17:51:59] [Startup Profile]
init_device_and_model: 66537.59ms (100.0%)
init_device_and_model.set_device: 0.04ms (0.0%)
init_device_and_model.init_distributed_environment: 12765.27ms (19.2%)
init_device_and_model.build_pipeline: 53758.28ms (80.8%)
init_device_and_model.build_pipeline.get_model_info: 429.33ms (0.8%)
init_device_and_model.build_pipeline.pipeline_instantiation: 53328.71ms (99.2%)
init_device_and_model.build_pipeline.pipeline_instantiation.load_modules: 53325.49ms (100.0%)
init_device_and_model.build_pipeline.pipeline_instantiation.load_modules.load_component.text_encoder: 3097.70ms (5.8%)
init_device_and_model.build_pipeline.pipeline_instantiation.load_modules.load_component.transformer: 983.16ms (1.8%)
init_device_and_model.build_pipeline.pipeline_instantiation.load_modules.load_component.vae: 177.89ms (0.3%)
init_device_and_model.build_pipeline.pipeline_instantiation.load_modules.load_component.tokenizer: 982.55ms (1.8%)
init_device_and_model.build_pipeline.pipeline_instantiation.load_modules.load_component.scheduler: 7.71ms (0.0%)
init_device_and_model.build_pipeline.pipeline_instantiation.initialize_pipeline: 0.32ms (0.0%)
init_device_and_model.build_pipeline.pipeline_instantiation.create_pipeline_stages: 1.81ms (0.0%)
```

Confirmed rank-0-only logging (single `[Startup Profile]` block in the log for this 1-GPU run). Generation itself completed successfully afterward (50/50 denoising steps, video saved to `outputs/`), so the profiler ran inline with a real end-to-end launch rather than in isolation.
